File: bin/make_fam2.py
#!/usr/bin/env python

import pandas as pd
import numpy as np
import sys

csv_file = sys.argv[1]
df = pd.read_csv(csv_file, sep=',', header=0)

# remove vcf column
df = df.drop(['vcf'], axis=1)
# make Family ID column
df['FID'] = np.arange(len(df)) + 1
# rename vcf with IID
df = df.rename({'sampleid': 'IID'}, axis=1)
# make paternal & maternal ID columns
df['PAT'] = 0
df['MAT'] = 0

# reorder columns
cols = list(df.columns.values)
new_cols = []
main_cols = ['FID','IID','PAT','MAT','SEX']
for col in main_cols:
    cols.remove(col)
    new_cols.append(col)

for col in cols:
    new_cols.append(col)

df = df[new_cols]

df.to_csv('sample.phe', sep=' ', index=False)

File: bin/make_fam2.R
#!/usr/bin/env Rscript

####################
# Import libraries #
####################

suppressPackageStartupMessages({
library(optparse)
library(data.table)
library(tidyverse)

    })

##########################################################
# Parse arguments                                        
##########################################################

option_list = list(
  make_option(c("--pheno_file"), action="store", default='None', type='character',
              help="String containing long format table for phenotypes."),
)

args = parse_args(OptionParser(option_list=option_list))

pheno_file          = args$pheno_file # file


########################################
### Data input
########################################
df = fread(pheno_file)

# remove vcf column
df = df %>% select(-vcf) %>% rename(IID=sampleid)
# make Family ID column
df$FID = df$IID

# make paternal & maternal ID columns
df$PAT = 0
df$MAT = 0

# reorder columns
df = df %>% select(FID,IID,PAT,MAT,SEX, everything())

write.table('sample.phe', sep=' ', quote=F, row.names=F)
